### PR TITLE
First form with translations

### PR DESCRIPTION
**Pull request by a6software**

### CODE REVIEW

File: incident-application-form/routes/your-details.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const express = require("express");
+const { body, validationResult } = require("express-validator");
+
+const router = express.Router();
+
+const template = "your-details";
+
+const languageCode = "cy";
+
+const i18n = require(`${__dirname}/../translations/${template}.${languageCode}.json`);
```
Comment (a6software) [inline incident-application-form/routes/your-details.js:10]: not sure how robust it is to require JSON this way, google opinions differ.

File: incident-application-form/routes/your-details.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+const title = i18n.title;
+
+router.get("/", function (req, res, next) {
+  res.render(template, { title, i18n });
+});
+
+router.post("/", [
+  body("notifier-type", "Notifier type is required").trim().notEmpty().escape(),
+  body("contact-name", "Contact name is required")
```
Comment (a6software) [inline incident-application-form/routes/your-details.js:20]: need translations etc, just a proof of concept really.

File: incident-application-form/translations/your-details.en.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "sections": {
+    "yourDetails": "Your Details"
+  },
+  "notifierType": {
+    "label":"Notifier Type",
+    "defaultValue": "Please choose an option",
+    "industry": "Industry",
+    "localAuthority":  "Local Authority"
+  }
+}
```
Comment (a6software) [inline incident-application-form/translations/your-details.en.json:12]: What I don't like about this approach is it's really easy to miss a key / value pair from one of the files. Or put in a typo. 

Maybe TypeScript could be used? A typed data structured here would mitigate this problem.

File: incident-application-form/views/your-details.njk [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    <fieldset>
+
+        <legend>{{ i18n.sections.yourDetails }}:</legend>
+
+        <label for="notifier-type">{{ i18n.notifierType.label }}:</label>
+
+        <select name="notifier-type" id="notifier-type">
+            <option value="">--{{ i18n.notifierType.defaultValue }}--</option>
+            <option value="industry">{{ i18n.notifierType.industry }}</option>
+            <option value="local-authority">{{ i18n.notifierType.localAuthority }}</option>
```
Comment (a6software) [inline incident-application-form/views/your-details.njk:15]: think these values will change - maybe this entire thing won't work.

Works as is, though.

File: incident-application-form/package.json [hunk 5]
```diff
@@ -3,14 +3,25 @@
...
     "express": "~4.16.1",
+    "express-validator": "^6.6.1",
     "http-errors": "~1.6.3",
     "jade": "~1.11.0",
-    "morgan": "~1.9.1"
+    "morgan": "~1.9.1",
+    "nunjucks": "^3.2.2"
+  },
+  "devDependencies": {
+    "jest": "^26.5.3",
```
Comment (a6software) [inline incident-application-form/package.json:21]: jest is in here as I added a couple of different approaches to translations - one more complex than the other. In the end I simplified and just required some JSON (as below).

File: incident-application-form/routes/your-details.js (hunk 1 above)
Comment (danWatkinson) [reply incident-application-form/routes/your-details.js:10]: I have no problems with it - i guess the alternative is to pop it out into a config file to map langaugeCode->.json but I think you can get too wrapped up in these things..

File: incident-application-form/translations/your-details.en.json (hunk 3 above)
Comment (danWatkinson) [reply incident-application-form/translations/your-details.en.json:12]: or mush it back into one json file?
industry: {
  "en": "Industry",
  "cy": "Yndustry",
}

less solid/dry, possibly easier to work with?

(i'm luke-warm on typescript; it has its upsides but I don't like introducing transpilation if i don't have to..)
(personal opinion; not going to get nailed to anything over it ;) )

File: incident-application-form/views/your-details.njk (hunk 4 above)
Comment (danWatkinson) [reply incident-application-form/views/your-details.njk:15]: 👍 if we end up going down the gov component route you barely write any html: https://design-system.service.gov.uk/components/select/

you just drop one of their macro's in and say "this is the label, this is the description, these are the options" and it just deals with all the ugly html itself.. so yeah, we don't want to think too hard about this yet; we'll revisit when we start trying to pull stuff from rainmaker api's

File: incident-application-form/package.json (hunk 5 above)
Comment (danWatkinson) [reply incident-application-form/package.json:21]: ❤️ jest
i'd gone with "if there's only one type of test i want it to be an e2e test" so got in and wired cypress up. if we had any more time i'd have been in there wiring jest in as well, but i certainly don't object to it. Not seeing any actual tests in ^ so not going to take this as a call to wire jest into the pipeline yet, but as soon as i see a test in there i'll start looking at the wiring ;)

**Comment by danWatkinson** [comment]
Looks good to me.. not merging in case it hides the comments ^^^ but whenever you look at this in the morning you have my blessing to merge it + it means i can pick it up out of master & get to work on cypress+ fleshing out the 1st page..

**Comment by danWatkinson** [comment]
I will take "continuous integration has not been set up" as a call to action though. will try and get us in a world of PR -> run cypress tests (+jest if needed) -> ok to merge to master. We don't have that on the project i was going to template on but i think we have all the pieces so.. i'll give it a crack. That'll be tomorrow evening though.

File: incident-application-form/translations/your-details.en.json (hunk 3 above)
Comment (a6software) [reply incident-application-form/translations/your-details.en.json:12]: I had it like this originally - I think I'm going to switch it back. Nice Welsh translation there :D

File: incident-application-form/views/your-details.njk (hunk 4 above)
Comment (a6software) [reply incident-application-form/views/your-details.njk:15]: Nice - should be easy enough to extract and use the govt components then. 

One concern I have with the govt tabs layout is using the tab headers to link to separate pages. Address that as and when, I guess.

File: incident-application-form/package.json (hunk 5 above)
Comment (a6software) [reply incident-application-form/package.json:21]: Yeah, as mentioned in chat, I had some tests but they were for the translator and I figured it was overkill really. Left Jest in as we may want / need it at some point.

**Pull request merged**